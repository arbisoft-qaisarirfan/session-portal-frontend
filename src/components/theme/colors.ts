export const colors = {
  gold: "#9f7d05",
  gray: "#908e8e",
  white: "#fff",
};

declare module "@mui/material/styles" {
  interface Palette {
    colors: typeof colors;
  }

  interface PaletteOptions {
    colors?: typeof colors;
  }
}
