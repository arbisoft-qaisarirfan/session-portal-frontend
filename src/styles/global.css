a {
    font-weight: var(--fontWeightNormal);
    color: var(--anchorColor);
    text-decoration: inherit;
}

a:hover {
    color: var(--assistColor);
}

button {
    border-radius: var(--buttonPrimary-border-radius);
    border: var(--buttonPrimary-border);
    padding: var(--buttonPrimary-padding);
    font-size: var(--buttonPrimary-font-size);
    font-weight: var(--buttonPrimary-font-weight);
    font-family: var(--buttonPrimary-font-family);
    background-color: var(--buttonPrimary-background-color);
    cursor: var(--buttonPrimary-cursor);
    transition: var(--buttonPrimary-transition);
}

button:hover {
    border-color: var(--assistColor);
}

button:focus,
button:focus-visible {
    outline: 4px auto -webkit-focus-ring-color;
}

/* Light Mode Styles */
@media (prefers-color-scheme: light) {
    :root {
        color: var(--secondaryColor);
        background-color: var(--whiteColor);
    }

    a:hover {
        color: var(--anchorColorLight);
    }

    button {
        background-color: var(--whiteColor);
    }
}
