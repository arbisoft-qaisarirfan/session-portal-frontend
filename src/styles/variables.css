:root {
    /* TYPOGRAPHY */
    --fontPrimary: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
    --fontSizeHeading: 3.2em;
    --fontSizeNormal: 16px;
    --fontWeightSmall: 200px;
    --fontWeightNormal: 400px;
  
    /* COLORS */
    /* TODO: replace these colors with actual colors */
    --assistColor: #508991;
    --assistColorLight: #eef3f4;
    --primaryColor: #004346;
    --whiteColor: #fff;
    --secondaryColor: #172a3a;
    --yellowColor: #f2c94c;
    --formLight: #b5b5b5;
    --ratingStarColor: #f2c94c;
    --color: rgba(255, 255, 255, 0.87);
    --backgroundColor: #242424;
    --anchorColor: #0000FF;
  
    /* Anchor Colors */
    --anchorColorHover: #FF6347;
    --anchorColorLight: #8b8bf6;
  
    /* LINE HEIGHTS */
    --lineHeightHeading: 2;
    --lineHeightNormal: 1.5;
    --lineHeightSmall: 1.15;
  
    /* Button Styles */
    --buttonPrimary-border-radius: 8px;
    --buttonPrimary-border: 1px solid transparent;
    --buttonPrimary-padding: 0.6em 1.2em;
    --buttonPrimary-font-size: 1em;
    --buttonPrimary-font-weight: var(--fontWeightNormal);
    --buttonPrimary-font-family: var(--fontPrimary);
    --buttonPrimary-background-color: var(--backgroundColor);
    --buttonPrimary-cursor: pointer;
    --buttonPrimary-transition: border-color 0.25s;
}