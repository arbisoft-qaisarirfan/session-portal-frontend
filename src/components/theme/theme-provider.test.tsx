import { createTheme } from "@mui/material/styles";
import Typography from "@mui/material/Typography";
import { render } from "@testing-library/react";

import { colors } from "./colors";
import InitColorSchemeScript from "./init-color-scheme-script";
import ThemeProvider from "./theme-provider";

import "@testing-library/jest-dom";

describe("ThemeProvider", () => {
  it("should render without crashing", () => {
    const { container } = render(
      <ThemeProvider>
        <div>Test</div>
      </ThemeProvider>
    );
    expect(container).toBeInTheDocument();
  });

  it("should render children components", () => {
    const { getByText } = render(
      <ThemeProvider>
        <div>Child Component</div>
      </ThemeProvider>
    );
    expect(getByText("Child Component")).toBeInTheDocument();
  });

  it("should apply the correct primary color", () => {
    const theme = createTheme();
    const { getByText } = render(
      <ThemeProvider customTheme={theme}>
        <div style={{ color: theme.palette.primary.main }}>Primary Color</div>
      </ThemeProvider>
    );
    expect(getByText("Primary Color")).toHaveStyle("color: #1976d2");
  });

  it("should apply the correct secondary color", () => {
    const theme = createTheme();
    const { getByText } = render(
      <ThemeProvider customTheme={theme}>
        <div style={{ color: theme.palette.secondary.main }}>Secondary Color</div>
      </ThemeProvider>
    );
    expect(getByText("Secondary Color")).toHaveStyle("color: #9c27b0");
  });

  it("should set typography font family correctly", () => {
    const theme = createTheme({
      typography: {
        fontFamily: ["Roboto Condensed ", "Roboto", "Helvetica", "Arial", "sans-serif"].join(),
      },
    });
    const { getByTestId } = render(
      <ThemeProvider customTheme={theme}>
        <Typography data-testid="test">Typography Test</Typography>
      </ThemeProvider>
    );
    expect(getByTestId("test")).toHaveStyle(`
      font-family: Roboto Condensed,Roboto,Helvetica,Arial,sans-serif
    `);
  });

  it("should ensure all shadows are set to 'none'", () => {
    const theme = createTheme();
    const shadows = [...theme.shadows].map(() => "none");

    expect(shadows.every((shadow) => shadow === "none")).toBe(true);
  });

  it("should have custom colors available", () => {
    expect(colors).toBeDefined();
  });

  it("should ensure correct contrast text for primary color", () => {
    const theme = createTheme();
    expect(theme.palette.primary.contrastText).toBe("#fff");
  });

  it("should ensure correct contrast text for secondary color", () => {
    const theme = createTheme();
    expect(theme.palette.secondary.contrastText).toBe("#fff");
  });

  it("should apply global styles correctly", () => {
    const theme = createTheme();
    const { container } = render(
      <ThemeProvider customTheme={theme}>
        <div>Global Style Test</div>
      </ThemeProvider>
    );
    expect(container.firstChild).toHaveStyle(`
      background-color: unset;
    `);
  });

  it("should handle light mode correctly", () => {
    const theme = createTheme({
      cssVariables: {
        colorSchemeSelector: "class",
      },
      colorSchemes: { light: true, dark: false },
      defaultColorScheme: "light",
    });
    expect(theme.palette.mode).toBe("light");
  });

  it("should handle dark mode correctly", () => {
    const theme = createTheme({
      cssVariables: {
        colorSchemeSelector: "class",
      },
      colorSchemes: { light: false, dark: true },
      defaultColorScheme: "dark",
    });
    expect(theme.palette.mode).toBe("dark");
  });

  it("should ensure background color is unset in global styles", () => {
    const { container } = render(
      <ThemeProvider>
        <div>Test</div>
      </ThemeProvider>
    );
    expect(container.firstChild).toHaveStyle("background-color: unset");
  });

  it("should set height to 100% for html, body, and __next root", () => {
    const { container } = render(
      <ThemeProvider>
        <div id="__next">
          <div>Test</div>
        </div>
      </ThemeProvider>
    );
    expect(container.firstChild).toHaveStyle("height: 100%");
  });

  it("should set width to 100% for html, body, and __next root", () => {
    const { container } = render(
      <ThemeProvider>
        <div id="__next">
          <div>Test</div>
        </div>
      </ThemeProvider>
    );
    expect(container.firstChild).toHaveStyle("width: 100%");
  });

  it("should ensure font size scales correctly in different screen sizes", () => {
    const theme = createTheme();
    expect(theme.typography.fontSize).toBeDefined();
  });

  it("should inherit Material UI theme values correctly", () => {
    const theme = createTheme();
    expect(theme.typography).toBeDefined();
  });

  it("should allow Material UI createTheme to override properties correctly", () => {
    const customTheme = createTheme({
      palette: {
        primary: { main: "#000000" },
      },
    });
    expect(customTheme.palette.primary.main).toBe("#000000");
  });

  it("should render correctly even when no children are provided", () => {
    const { container } = render(<ThemeProvider />);
    expect(container).toBeInTheDocument();
  });

  it("should not apply any invalid styles", () => {
    const { container } = render(
      <ThemeProvider>
        <div>Test</div>
      </ThemeProvider>
    );
    expect(container).not.toHaveStyle("invalidProperty: none");
  });
});

describe("InitColorSchemeScript", () => {
  it("should render without crashing", () => {
    const { container } = render(<InitColorSchemeScript />);
    expect(container).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<InitColorSchemeScript />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render InitColorSchemeScriptMui internally", () => {
    const { container } = render(<InitColorSchemeScript />);
    expect(container.firstChild).toBeDefined();
  });

  it("should not apply unexpected props", () => {
    const { container } = render(<InitColorSchemeScript />);
    expect(container.firstChild).not.toHaveAttribute("unexpected-prop");
  });
});
