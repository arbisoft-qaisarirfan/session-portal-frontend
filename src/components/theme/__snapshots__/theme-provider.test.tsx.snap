// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitColorSchemeScript should match the snapshot 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    (function() {
try {
  let colorScheme = '';
  const mode = localStorage.getItem('mui-mode') || 'system';
  const dark = localStorage.getItem('mui-color-scheme-dark') || 'dark';
  const light = localStorage.getItem('mui-color-scheme-light') || 'light';
  if (mode === 'system') {
    // handle system mode
    const mql = window.matchMedia('(prefers-color-scheme: dark)');
    if (mql.matches) {
      colorScheme = dark
    } else {
      colorScheme = light
    }
  }
  if (mode === 'light') {
    colorScheme = light;
  }
  if (mode === 'dark') {
    colorScheme = dark;
  }
  if (colorScheme) {
    document.documentElement.classList.remove('%s'.replace('%s', light), '%s'.replace('%s', dark));
      document.documentElement.classList.add('%s'.replace('%s', colorScheme));document.documentElement.setAttribute('.%s', colorScheme);
  }
} catch(e){}})();
  </script>
</DocumentFragment>
`;
